zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -58,4 +170,62 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_founders_address() -> Result<(), Report> {
+        founders_address_for_network(Network::Mainnet)?;
+        founders_address_for_network(Network::Testnet)?;
+
+        Ok(())
+    }
+
+    fn founders_address_for_network(network: Network) -> Result<(), Report> {
+        // Test if all the founders addresses are paid in the entire period
+        // where rewards are active.
+        let mut populated_addresses: Vec<Address> = Vec::new();
+        let mut hardcoded_addresses = FOUNDERS_REWARD_ADDRESSES_MAINNET;
+
+        if network == Network::Testnet {
+            hardcoded_addresses = FOUNDERS_REWARD_ADDRESSES_TESTNET;
+        }
+
+        // After blossom, founders addresses will change at a greater interval but we should catch
+        // all the changes if we increment by founders_address_change_interval() for
+        // the entire period where founders reward are paid.
```

Comment:
I'm not sure this comment is correct. Why should that *necessarily* catch all the changes? I think it actually does, but partly by a coincidence of when the Blossom and Canopy activations happened. You could in principle miss the last address, because that was used for less than one (post-Blossom) period, and the heights you're checking are out of sync with the address changes after Blossom activation.

Reply:
If we just increment by 1 block we will make sure we catch everything whatever mix of upgrades the network could potentially have however this will make the test run for a pretty long time. It will take around 4 mins 40 secs on my laptop to do the test for the mainnet and the testnet.

@teor2345 what do you think. iirc we have some policy for long tests however not sure where we are about that this days.

Reply:
It's not necessary to check every block, but my point is that the current test code is depending on a coincidence of when Blossom and Canopy were activated. It would be possible to calculate the exact address change heights and then check the first and last address within each period.

Reply:
I changed to every block by now and i added a follow up ticket for this so we dont block the entire PR with this issue. https://github.com/ZcashFoundation/zebra/issues/1310